app/services/DriveService.scala

```diff
@@ -34,34 +35,90 @@ object DriveService {
-        }.toOption.getOrElse("")
-        val exportLinks = Try {
-          file.getExportLinks.asScala.toMap
-        }.toOption.getOrElse(Map.empty)
-
-        File(
-          id = file.getId,
-          title = file.getName,
-          output = file.getMimeType,
-          outputPreview = file.getWebViewLink, // TODO - returning editable links - need to process to preview
-          outputDownload = file.getWebContentLink, // TODO - returning nulls
-          outputThumbnail = thumbnailLink, // TODO - returning 404s
-          exportLinks = exportLinks,
-          customProperties = customProps
-        )
-      })
+      .map(file =>
+        // transformToResearchRecord(extractRawFile(file)) // define functions with def
+        (extractRawFile andThen transformToResearchRecord)(file) // define functions with val
+    )
+  }
+
+  val extractRawFile = (fileStream: File) => {
+    val customProps = Try {
+      fileStream.getProperties.asScala.toMap
+    }.toOption.getOrElse(Map.empty)
+    val exportLinks = Try {
+      fileStream.getExportLinks.asScala.toMap
+    }.toOption.getOrElse(Map.empty)
+
+    DriveFile(
+      id = fileStream.getId,
+      title = fileStream.getName,
+      output = fileStream.getMimeType,
+      outputPreview = fileStream.getWebViewLink,
+      outputDownload = fileStream.getWebContentLink, // TODO - returning nulls
+      exportLinks = exportLinks,
+      customProperties = customProps
+    )
+  }
+
+  val transformToResearchRecord = (file: DriveFile) => {
+    ResearchRecord.apply(
+      id = file.id,
+      title = file.title,
+      team = file.customProperties.getOrElse("team", "Unknown Team"),
+      output = setOutputType(file.output),
+      outputUrl = convertToPreviewLink(file.outputPreview),
+      outputThumbnail = createThumbnailLink(file.id),
+      outputDownload = file.outputDownload, // TODO returning null - permissions issue?
+      okr = file.customProperties.getOrElse("okr", ""),
+      year = file.customProperties.getOrElse("year", "0").toInt,
+      quarter = setQuarter(file.customProperties.getOrElse("quarter", "Unknown Quarter")),
+      tags = setRandomTags(file.customProperties.getOrElse("tags", "No tags found"))
+    )
+  }
+
+  def convertToPreviewLink(str: String) = {
+    str.replaceAll("edit", "preview")
+  }
+
+  def createThumbnailLink(id: String): String = {
+    s"https://drive.google.com/thumbnail?authuser=0&sz=w320&id=$id"
+  }
+
+  // TODO temporary measure to hard gcode random tags
+  def setRandomTags(str: String): List[Tag] = {
+    str match {
+      case "No tags found" => Tag.selectRandomTags()
+      case _ => Nil
+    }
+  }
+
+  def setQuarter(str: String): Quarter = {
+    str match {
+      case "1" => Q1
+      case "2" => Q2
+      case "3" => Q3
+      case "4" => Q4
+      case "Q1" => Q1
+      case "Q2" => Q2
+      case "Q3" => Q3
+      case "Q4" => Q4
+      case "Unknown Quarter" => UnknownQuarter
+      case _ => UnknownQuarter
+    }
+  }
```

Comment:
since this basically a string->quarter constructor, I would personally make a companion object in the same file as the Quarter stuff, and put it in there.
```scala
object Quarter {
  def fromLabel(str: String): Quarter = {
    ...
  }
}
```